services/SignupMetricsService.ts

```diff
@@ -25,21 +25,61 @@ export interface SignupCountResult {
 }
 
 export default class SignupMetricsService {
-  private tableName: string = process.env.DYNAMODB_TABLE || DEFAULT_TABLE;
-  private dynamoService: DynamoService;
+  private readonly tableName: string = process.env.DYNAMODB_TABLE ?? DEFAULT_TABLE;
+
+  private readonly dynamoService: DynamoService;
 
   public constructor(dynamoService: DynamoService) {
     this.dynamoService = dynamoService;
   }
 
+  private static buildFilterParams(options: SignupQueryOptions): FilterParams {
+    let filterParams = {};
+    if (options.startDate && options.endDate) {
+      filterParams = {
+        ExpressionAttributeValues: {
+          ':endDate': options.endDate.toISOString(),
+          ':startDate': options.startDate.toISOString(),
+        },
+        FilterExpression: 'createdAt BETWEEN :startDate AND :endDate',
+      };
+    } else if (options.startDate) {
+      filterParams = {
+        ExpressionAttributeValues: {
+          ':startDate': options.startDate.toISOString(),
+        },
+        FilterExpression: 'createdAt >= :startDate',
+      };
+    } else if (options.endDate) {
+      filterParams = {
+        ExpressionAttributeValues: {
+          ':endDate': options.endDate.toISOString(),
+        },
+        FilterExpression: 'createdAt <= :endDate',
+      };
+    }
+
+    return filterParams;
+  }
+
+  private static mapItemsToSignups(items: AttributeMap[]): Signup[] {
+    return items.map(item => ({
+      /* eslint-disable @typescript-eslint/no-base-to-string */
+      apis: item.apis.toString(),
```

Comment:
not relevant for this ticket, but we should probably use https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB/Converter.html instead of toString

Reply:
Agreed, but looks like a heavier lift than just switching out... doing a little testing and using the `Converter.output` is breaking unit and integration tests and it looks like the items it receives are just plain object, not DynamoDB AttributeValues like the types say it should be.